api/mod/src/database/database.py

```diff
@@ -0,0 +1,23 @@
+"""
+This module defines database config and access methods
+"""
+
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from mod.src.settings import SETTINGS
+
+Base = declarative_base()
+
+engine = create_engine(SETTINGS.database_url, connect_args={"check_same_thread": False})
+SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+
+def get_db() -> sessionmaker:
```

Comment:
Yes I think it creates a new connection for every request, but I think its common practice to have a new session for each request, isnt it?

Reply:
It is definitely common to use one transaction per request. But from my (little) experience, you would rather reuse the connection. Just think about a MySQL database or whatever may be used in the future. Is it really desirable to authenticate against the database (which includes logging in, i.e. transmitting db admin username/password, which is rather time consuming) for every single request?

Reply:
I think my problem is that I have no clue what a "session" should be in the database world

Reply:
When using the orm style of sqlalchemy (which we do), the session is our communication point to the database (instead of a connection). In the background, the session also requests a connection from the connection pool as soon as queries are issued. The session additionally keeps track of all orm objects we use within that session and automatically keeps them in sync with the database. Apart from that, the session behaves like a normal connection regarding transactions, committing, flushing and so on. For further information: https://docs.sqlalchemy.org/en/14/orm/session_basics.html

Reply:
So to come back to the original question yes, it creates a new session for each request. This session is then the context holder for all the database interactions that happen during this request. And this context should be for that request only and every other request should start with a new clean context. So as far as I understand, this is the way it should be used ^^